app/models/tag.rb

```diff
@@ -15,6 +15,8 @@ class Tag < ApplicationRecord
   scope :is_category, -> { joins(:classification).merge(Classification.is_category) }
   scope :is_entry,    -> { joins(:classification).merge(Classification.is_entry) }
 
+  validates :name, :unique_within_region => true
```

Comment:
Do we need a migration to fix any existing duplicates?

Reply:
this was not introduced, this is the existing logic (it may be broken though)